PRINCIPLES.md

```diff
@@ -0,0 +1,254 @@
+
+<!--
+- If the software agents fail to bring the system's state in line with its desired state, a human operator is notified.
+- Software agents continuously check that the running system under management matches the desired state configuration, and if it does not, immediately either take remedial action to bring the system back in line with stated expectations or, if this cannot be done, alert a human operator that the system is no longer meeting expectations
+- Automated delivery: Delivery of the declarative descriptions, from the repository to runtime environment, is fully automated.
+- Software Agents: Reconcilers maintain system state and apply the resources described in the declarative configuration.
+- Closed loop: Actions are performed on divergence between the version controlled declarative configuration and the actual state of the target system.
+- This automation should serve its human users.
+  To this point, reconciliation automation for GitOps should include a way for humans to temporarily take back the reins as needed.
+- Implies observability. State of the running system must be observable. Ideally in same format as the configuration so we can compare like to like. 
+- >I agree. I think of this as an observability requirement and tend to say that "the current state of a running system must be observable" to mean the same thing. This is, in my opinion, required for state reconciliation. I didn't add a note there explicitly though, as that section was unfinished.
+-->
+
+
+### 4. Operations through declaration
+
+<div style="background: #E3F2FD; border-left: 3px solid #0D47A1; padding:10px; margin-bottom:5px; font-style:italic">
+When wishing to operate on a software system, a human or software agent will not interact with the running system and modify it directly.
+Instead, the agent will create a new declarative version of the desired state in the state store.
+</div>
+<!--
+- All normal operations should occur via the creation of a new uniquely named version, not through direct interaction ith the system under management
+- Break glass - exceptions are acceptable and quite likely. and credentials to access and mutate the system must still be available in almost all cases.
+-->
+
+<!--
+## General notes on the GitOps principles
+
+- Tool and system agnosticicsm.
+  
+- It's ok to be pragmatic with implementations - We recognise that few real systems can be “100% GitOps”.
+  Real world systems have a lifetime measured in years and must interact with many other systems created under alternative paradigms.
+  As such they involve compromises and workarounds.
+  
+- lowest turtle isn't declarative, because the real world isn't.
+  We aim for the lowest turtle to be as small as possible (imperative surface area )
+
+- Git as a satte store: - the "state store" can be a subfolder in a particular branch, and repo.
+  Principles only applies to the data used to configure a running system, not other data in the same git repository.
```

Comment:
```suggestion
  Principles only applies to the data used to configure a running system, not other data in the same git repository nor runtime application data.
```

Aligned with one of my comments above, GitOps is not about backup nor disaster recovery.

Reply:
Agreed. DR includes application data, GitOps doesn't.